checkers.go

```diff
@@ -293,24 +293,33 @@ func (checker *matchesChecker) Check(params []interface{}, names []string) (resu
 }
 
 func matches(value, regex interface{}) (result bool, error string) {
-	reStr, ok := regex.(string)
-	if !ok {
-		return false, "Regex must be a string"
-	}
-	valueStr, valueIsStr := value.(string)
-	if !valueIsStr {
-		if valueWithStr, valueHasStr := value.(fmt.Stringer); valueHasStr {
-			valueStr, valueIsStr = valueWithStr.String(), true
-		}
-	}
-	if valueIsStr {
-		matches, err := regexp.MatchString("^"+reStr+"$", valueStr)
+	var matcher *regexp.Regexp
+	switch r := regex.(type) {
+	case string:
+		m, err := regexp.Compile("^"+r+"$")
 		if err != nil {
 			return false, "Can't compile regex: " + err.Error()
+		} else {
+			matcher = m
```

Comment:
Minor style nit: since the if returns the `else {` isn't needed. (Govent or Golint is built in to my editor and has somewhat ingrained this practice in my head. I don't actually feel strongly about it though)